pkg/plan/down.go

```diff
@@ -23,20 +34,24 @@ func (p *Plan) Down(ctx context.Context) error {
 		}
 	}()
 
+	nodesChan := dependenciesGraph.Run()
+
 	wg := parallel.NewWaitGroup()
 	wg.Add(len(p.body.Releases))
 
-	for i := range p.body.Releases {
-		go func(ctx context.Context, wg *parallel.WaitGroup, rel release.Config) {
+	for node := range nodesChan {
+		go func(ctx context.Context, wg *parallel.WaitGroup, node *dependency.Node[release.Config]) {
 			defer wg.Done()
+			rel := node.Data
 			_, err := rel.Uninstall(ctx)
+			node.SetSucceeded()
```

Comment:
Do we really want to always continue uninstall even if some dependency has failed? I think we should call either `node.SetSucceeded()` or `node.SetFailed()` like during install to halt uninstall if anything fails

Reply:
I have spent several time in figuring this out and I would be needing help with this

So in my first draft I used `node.SetFailed()` but then the channel is stuck and no new nodes are fetched and the program never ends.

I am not sure why this happens, since I think this is my first contribution to a go code, I might be missing something. I tried debugging in several ways and still was not able to figure out the root cause of the issue, while this thing perfectly works in case when we are bringing the stack up.

Reply:
Is there an example config for which it's stuck? Cause tests are working perfectly fine with `node.SetFailed()` properly called